src/auspex/instruments/X6.py

```diff
@@ -50,22 +50,30 @@ def __init__(self, settings_dict=None):
         self.channel_tuple  = (1,0,0)
 
         self.dtype = np.float64
+        self.ideal_data = None
 
         if settings_dict:
             self.set_all(settings_dict)
 
     def set_all(self, settings_dict):
         for name, value in settings_dict.items():
-
-            if name == "kernel" and isinstance(value, str) and value:
-                #assume that the kernel is saved as a complex array
-                self.kernel = np.loadtxt(os.path.join(config.KernelDir, value+'.txt'), dtype=complex, converters={0: lambda s: complex(s.decode().replace('+-', '-'))})
+            if name == "kernel" and value:
+                #check if the kernel is given as an existing path or an expression to eval
+                if os.path.exists(os.path.join(config.KernelDir, value+'.txt')):
+                    self.kernel = np.loadtxt(os.path.join(config.KernelDir, value+'.txt'), dtype=complex, converters={0: lambda s: complex(s.decode().replace('+-', '-'))})
+                else:
+                    try:
+                        self.kernel = eval(value)
+                    except:
+                        raise ValueError('Kernel invalid. Provide a file name or an expression to evaluate')
             elif name == "kernel_bias" and isinstance(value, str) and value:
                 self.kernel_bias = eval(value)
             #elif hasattr(self, name):
             #        setattr(self, name, value)
             elif name == "channel":
                 setattr(self, 'phys_channel', int(value))
+            elif name == 'ideal_fake_data': # for testing purposes
```

Comment:
Looks like the configurations below use "ideal_data" not "ideal_fake_data"?

Reply:
Yes. In reality, I'm only using the X6 property for now, not the channel. But may be useful later for two-qubit gates...